Detectors/TPC/calibration/src/CalibPadGainTracks.cxx

```diff
@@ -43,8 +43,9 @@ void CalibPadGainTracks::processTracks()
     refit = std::make_unique<o2::gpu::GPUO2InterfaceRefit>(mClusterIndex, mFastTransform.get(), mField, mTPCTrackClIdxVecInput->data(), mClusterShMapTPC);
   }
 
-  for (const auto& trk : *mTracks) {
-    processTrack(trk, refit.get());
+  const int loopEnd = (nMaxTracks < 0) ? mTracks->size() : ((nMaxTracks > mTracks->size()) ? mTracks->size() : nMaxTracks);
+  for (int i = 0; i < loopEnd; ++i) {
+    processTrack((*mTracks)[i], refit.get());
```

Comment:
@davidrohr , is there any ordering in the tracks, e.g. momenta. Otherwise random sampling would be better.

Reply:
the tracks should be roughly ordered by time. Perhaps in any case random sampling would be better. You can use any simple fast linear congruence generator for that, should have almost zero overhead, except for more random memory reads.

Reply:
okay, then I change that